packages/pg-gateway/src/connection.ts

```diff
@@ -417,19 +428,19 @@ export default class PostgresConnection {
    */
   async completeAuthentication() {
     this.isAuthenticated = true;
+
     this.sendAuthenticationOk();
 
+    this.socket.pause();
+    await this.options.onAuthenticated?.(this.state);
+    this.socket.resume();
```

Comment:
we call `onAuthenticated` as soon as we send the `AuthenticationOk` message, but before sending the `ReadyForQuery` message so we are certain that the client won't send new messages before the hook is executed.